src/vulkan/buffer_backed_descriptor.cc

```diff
@@ -45,7 +45,10 @@ Result BufferBackedDescriptor::RecordCopyDataToResourceIfNeeded(
   for (size_t i = 0; i < resources.size(); i++) {
     if (!buffers[i]->ValuePtr()->empty()) {
       resources[i]->UpdateMemoryWithRawData(*buffers[i]->ValuePtr());
-      buffers[i]->ValuePtr()->clear();
+      // Keep the data if the resource is read only. Otherwise the data will
+      // be regenerated after a draw.
```

Comment:
The "regenerated" word confuses me. Maybe just say:

If the resource is read-only, keep the buffer data; Amber won't copy read-only resources back into the host buffers, so it makes sense to leave the buffer intact.